src/BaseServer.ts

```diff
@@ -132,4 +132,249 @@ export abstract class BaseServer {
+            }
+            const id = Number.parseInt(matches[3]);
+            if (request.method === 'GET') {
+                try {
+                    return {
+                        status: 200,
+                        body: this.getOne(name, id, params),
+                        headers: {
+                            'Content-Type': 'application/json',
+                        },
+                    };
+                } catch (error) {
+                    return {
+                        status: 404,
+                        headers: {},
+                    };
+                }
+            }
+            if (request.method === 'PUT') {
+                try {
+                    if (request.requestJson == null) {
+                        return {
+                            status: 400,
+                            headers: {},
+                        };
+                    }
+                    return {
+                        status: 200,
+                        body: this.updateOne(name, id, request.requestJson),
+                        headers: {
+                            'Content-Type': 'application/json',
+                        },
+                    };
+                } catch (error) {
+                    return {
+                        status: 404,
+                        headers: {},
+                    };
+                }
+            }
+            if (request.method === 'PATCH') {
+                try {
+                    if (request.requestJson == null) {
+                        return {
+                            status: 400,
+                            headers: {},
+                        };
+                    }
+                    return {
+                        status: 200,
+                        body: this.updateOne(name, id, request.requestJson),
+                        headers: {
+                            'Content-Type': 'application/json',
+                        },
+                    };
+                } catch (error) {
+                    return {
+                        status: 404,
+                        headers: {},
+                    };
+                }
+            }
+            if (request.method === 'DELETE') {
+                try {
+                    return {
+                        status: 200,
+                        body: this.removeOne(name, id),
+                        headers: {
+                            'Content-Type': 'application/json',
+                        },
+                    };
+                } catch (error) {
+                    return {
+                        status: 404,
+                        headers: {},
+                    };
+                }
+            }
+        }
+        return {
+            status: 404,
+            headers: {},
+        };
+    }
 }
+
+type BaseRequest = {
+    url?: string;
+    method?: string;
+    requestJson?: Record<string, any> | undefined;
```

Comment:
Not sure what `requestJson` means. Isn't it the `body`?

Reply:
I kept the old code to avoid renaming too many things. You should remember! ;)